### PR TITLE
Dijkstra Algorithm

### PR DESCRIPTION
**Pull request by kapildeshpande**
Add dijkstra's algorithm

### CODE REVIEW

File: Graph algorithms/Dijkstra_algorithm.cpp [hunk 1]
```diff
@@ -0,0 +1,71 @@
+//Dijkstra algorithm using adjacency list representation of graph
+// used for finding shortest path in a graph from one vertex to all other vertex
+//Time complexity O(n + m)
+//space complexity O()
+
+#include<bits/stdc++.h>
```
Comment (faheel) [inline Graph algorithms/Dijkstra_algorithm.cpp:6]: Please use only those header files which are required.

File: Graph algorithms/Dijkstra_algorithm.cpp [hunk 2]
```diff
@@ -0,0 +1,71 @@
+//Dijkstra algorithm using adjacency list representation of graph
+// used for finding shortest path in a graph from one vertex to all other vertex
+//Time complexity O(n + m)
+//space complexity O()
```
Comment (faheel) [inline Graph algorithms/Dijkstra_algorithm.cpp:4]: Space complexity is incomplete.
Also, please format these comments as mentioned in the guidelines.

File: Graph algorithms/Dijkstra_algorithm.cpp [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+	}
+	cout<<"Enter source vertex\n";
+	cin>>s;
+	dijkstra(s);
+	cout<<"Minimum distence from source vertex to all other vertex is \n";
+	for (int i=1;i<=n;i++) {
+		cout<<dist[i]<<" ";
+	}
+	cout<<"\n";
+
```
Comment (faheel) [inline Graph algorithms/Dijkstra_algorithm.cpp:70]: Missing `return` statement.

File: Graph algorithms/Dijkstra_algorithm.cpp [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+				s.insert(make_pair(dist[adj[x][i].second],adj[x][i].second));
+			}
+		}
+
+	}
+}
+
+int main () {
+
+	int n,m,s;
```
Comment (faheel) [inline Graph algorithms/Dijkstra_algorithm.cpp:51]: Use better variable names for all 3 of these.

File: Graph algorithms/Dijkstra_algorithm.cpp [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+	cout<<"Enter no. of vertex and edges in graph ";
+	cin>>n>>m;
+	initialise(n);
+	cout<<"Enter edges \n";
+	while (m--) {
+		int a,b,c;
```
Comment (faheel) [inline Graph algorithms/Dijkstra_algorithm.cpp:57]: Use better variable names for all 3 of these too.

File: Graph algorithms/Dijkstra_algorithm.cpp [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+
+#define MAX 10001
+
+vector < pair <int,int> > adj[MAX];
+bool visited[MAX];
+int dist[MAX];
+
+void initialise (int n) {
+
+	for (int i=0;i<=n;i++) {
```
Comment (faheel) [inline Graph algorithms/Dijkstra_algorithm.cpp:17]: Use better variable names for `n` and `i`.

**Review by faheel — CHANGES_REQUESTED** [review]
- Please indent your code using 4 spaces per level.
- Remove the blank line right after each function's header.

**Comment by kapildeshpande** [comment]
Ok I will correct all that issues.

**Comment by faheel** [comment]
Branch was deleted. Continued in #78.

**Pull request closed without merging**